src/Classifier.php

```diff
@@ -9,28 +9,67 @@
 class Classifier
 {
     /**
-     * @var array
+     * @var ?callable(string): array<int, string>
+     */
+    private $tokenizer;
+
+    /**
+     * @var array<string, array<string, int>>
      */
     private array $words = [];
 
     /**
-     * @var array
+     * @var array<string, int>
      */
     private array $documents = [];
 
+    private bool $uneven = false;
+
+    /**
+     * @param callable(string): array<int, string> $tokenizer
+     */
+    public function __construct(?callable $tokenizer = null)
```

Comment:
I wouldn't expect the tokenizer to be passed as the very first argument. 
Let's change its passing to a method, what do you think about it?

```php
$classifier = new Classifier();

$classifier->setTokenizer(function($string) {
  return ...
});
```

Reply:
sure, why not